x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/saml/idp/CloudIdp.java

```diff
@@ -0,0 +1,121 @@
+import static org.elasticsearch.xpack.idp.IdentityProviderPlugin.IDP_SSO_REDIRECT_ENDPOINT;
+
+public class CloudIdp implements SamlIdentityProvider {
+
+    private final String entityId;
+    private final HashMap<String, String> ssoEndpoints = new HashMap<>();
+    private final HashMap<String, String> sloEndpoints = new HashMap<>();
+    private final X509Credential signingCredential;
+
+    public CloudIdp(Environment env, Settings settings) {
+        this.entityId = require(settings, IDP_ENTITY_ID);
+        this.ssoEndpoints.put("redirect", require(settings, IDP_SSO_REDIRECT_ENDPOINT));
+        this.ssoEndpoints.computeIfAbsent("post", v -> settings.get(IDP_SSO_POST_ENDPOINT.getKey()));
+        this.sloEndpoints.computeIfAbsent("post", v -> settings.get(IDP_SLO_POST_ENDPOINT.getKey()));
+        this.sloEndpoints.computeIfAbsent("redirect", v -> settings.get(IDP_SLO_REDIRECT_ENDPOINT.getKey()));
+        this.signingCredential = buildSigningCredential(env, settings);
+    }
+
+    @Override
+    public String getEntityId() {
+        return entityId;
+    }
+
+    @Override
+    public String getSingleSignOnEndpoint(String binding) {
+        return ssoEndpoints.get(binding);
+    }
+
+    @Override
+    public String getSingleLogoutEndpoint(String binding) {
+        return sloEndpoints.get(binding);
+    }
+
+    @Override
+    public X509Credential getSigningCredential() {
+        return signingCredential;
+    }
+
+    private static String require(Settings settings, Setting<String> setting) {
+        if (settings.hasValue(setting.getKey())) {
+            return setting.get(settings);
+        } else {
+            throw new SettingsException("The configuration setting [" + setting.getKey() + "] is required");
+        }
+    }
+
+    static X509Credential buildSigningCredential(Environment env, Settings settings) {
+        final X509KeyPairSettings keyPairSettings = X509KeyPairSettings.withPrefix("xpack.idp.signing.", false);
+        final X509KeyManager keyManager = CertParsingUtils.getKeyManager(keyPairSettings, settings, null, env);
+        if (keyManager == null) {
+            return null;
+        }
+        final Set<String> aliases = new HashSet<>();
+        final String configAlias = IDP_SIGNING_KEY_ALIAS.get(settings);
+        if (Strings.isNullOrEmpty(configAlias)) {
+            final String[] rsaAliases = keyManager.getServerAliases("RSA", null);
+            if (null != rsaAliases) {
+                aliases.addAll(Arrays.asList(rsaAliases));
+            }
+            final String[] ecAliases = keyManager.getServerAliases("EC", null);
+            if (null != ecAliases) {
+                aliases.addAll(Arrays.asList(ecAliases));
+            }
+            if (aliases.isEmpty()) {
+                throw new IllegalArgumentException("The configured keystore for xpack.idp.signing.keystore does not contain any RSA or EC" +
+                    " key pairs");
```

Comment:
Nit: I prefer putting the whole string on a new line rather than splitting it. I find it easier to check that the message has the right spaces, and no duplicate words, etc if I can read it in 1 unit.
```
    throw new IllegalArgumentException(
        "The configured keystore for xpack.idp.signing.keystore does not contain any RSA or EC key pairs");
```